laiaaa0/assignment1/q1.cpp

```diff
@@ -1,13 +1,25 @@
 #include <iostream>
 #include <vector>
+#include <ctype.h>
+#include <algorithm>
 using namespace std;
-
-bool anagrams_caseSensitive (string & s1, string & s2){
+/*
+Possible solutions
+option 1 )
+  order the letters in the word. and compare
+  cost : O(nlogn)
+option 2 )
+  count characters (128 or 256?
+  cost : O (n)
+  )
+*/
+bool anagrams_caseSensitive (const string & s1, const string & s2){
   // Compare lengths. If they are not the same, return false.
   if (s1.size() != s2.size()) return false;
 
-  // Create a vector of the size of the ascii table.
-  std::vector<int> counts(128);
+  // Create a vector of the size of the EXTENDED ascii table.
```

Comment:
I don't want to insist much on this, just remember that it's "a bit" more complex than that. :)
Did you try running it and entering non-ascii characters?